agent/config/parse_linux_test.go

```diff
@@ -0,0 +1,63 @@
+//go:build linux && unit
+// +build linux,unit
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package config
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestParseGMSACapabilitySupported(t *testing.T) {
+	os.Setenv("ECS_GMSA_SUPPORTED", "True")
```

Comment:
nit: something I learnt recently - you can use `t.Setenv` to set the env variable. it would cleanup as part of the test cleanup workflow without explicit defer statements. 

https://pkg.go.dev/testing#B.Setenv

applicable to multiple tests in this PR

Reply:
made changes to use t.setenv